src/rust/ide/lib/ast/impl/src/lib.rs

```diff
@@ -303,6 +310,32 @@ impl Ast {
         //   Still, no need at this point.
         self.iter_recursive().find(|ast| ast.repr() == repr)
     }
+
+    /// Get the index (relative to self) for given child node.
+    pub fn index_of_child(&self, child:&Ast) -> FallibleResult<Index> {
+        let searched_token  = Token::Ast(&child);
+        let mut found_child = false;
+        let mut position    = 0;
+        self.shape().feed_to(&mut |token:Token| {
+            if searched_token == token {
+                found_child = true
+            } else if !found_child {
+                position += token.len()
+            }
+        });
+        if found_child {
+            Ok(Index::new(position))
+        } else {
+            Err(NoSuchChild.into())
+        }
+    }
```

Comment:
`Index` term in this context is really misleading for me. It's not natural that this means "index of character". Usually we use rather "offset" term.

Reply:
-> child_offset